http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AzureServiceInterp.scala

```diff
@@ -0,0 +1,351 @@
+          )
+      }
+
+      controlledResourceOpt <- controlledResourceQuery
+        .getResourceTypeForRuntime(runtime.id, WsmResourceType.AzureVm)
+        .transaction
+      azureRuntimeControlledResource <- F.fromOption(
+        controlledResourceOpt,
+        AzureRuntimeControlledResourceNotFoundException(cloudContext, runtimeName, context.traceId)
+      )
+
+      hasPermission <- authProvider.hasPermission(WsmResourceSamResourceId(azureRuntimeControlledResource.resourceId),
+                                                  WsmResourceAction.Write,
+                                                  userInfo)
+
+      _ <- context.span.traverse(s => F.delay(s.addAnnotation("Done auth call for delete azure runtime permission")))
+      _ <- if (hasPermission) F.unit
+      else F.raiseError[Unit](RuntimeNotFoundException(cloudContext, runtimeName, "permission denied"))
+
+      _ <- if (runtime.status.isDeletable) F.unit
+      else F.raiseError[Unit](RuntimeCannotBeDeletedException(cloudContext, runtime.clusterName, runtime.status))
+
+      _ <- clusterQuery.markPendingDeletion(runtime.id, context.now).transaction
+
+      // For now, azure disk life cycle is tied to vm life cycle and incompatible with disk routes
+      _ <- persistentDiskQuery.markPendingDeletion(diskId, context.now).transaction
+
+      _ <- publisherQueue.offer(
+        DeleteAzureRuntimeMessage(runtime.id,
+                                  diskId,
+                                  workspaceId,
+                                  azureRuntimeControlledResource.resourceId,
+                                  Some(context.traceId))
+      )
+    } yield ()
+
+  private[service] def convertToDisk(userInfo: UserInfo,
+                                     cloudContext: CloudContext,
+                                     diskName: DiskName,
+                                     config: PersistentDiskConfig,
+                                     req: CreateAzureRuntimeRequest,
+                                     now: Instant): Either[Throwable, PersistentDisk] = {
+    // create a LabelMap of default labels
+    val defaultLabelMap: LabelMap =
+      Map(
+        "diskName" -> diskName.value,
+        "cloudContext" -> cloudContext.asString,
+        "creator" -> userInfo.userEmail.value
+      )
+
+    // combine default and given labels
+    val allLabels = req.azureDiskConfig.labels ++ defaultLabelMap
+
+    for {
+      // check the labels do not contain forbidden keys
+      labels <- if (allLabels.contains(includeDeletedKey))
+        Left(IllegalLabelKeyException(includeDeletedKey))
+      else
+        Right(allLabels)
+    } yield PersistentDisk(
+      DiskId(0),
+      cloudContext,
+      ZoneName(req.region.toString),
+      diskName,
+      userInfo.userEmail,
+      //TODO: should this be something else?
+      PersistentDiskSamResourceId("fakeUUID"),
```

Comment:
I think WSM probably creates a sam resource for this. Not sure if this info is returned from WSM, but I think we probably do need this for `getDisk` API

Reply:
We talked about punting on disk interoperability because of various sam/service account concerns. Tracked in this ticket https://broadworkbench.atlassian.net/browse/IA-3213